ci-operator/step-registry/cucushift/installer/check/azure/usertags/cucushift-installer-check-azure-usertags-commands.sh

```diff
@@ -0,0 +1,253 @@
+        oc patch infrastructure cluster -p "[{\"op\":\"replace\",\"path\":\"/status/platformStatus/azure/resourceTags\",\"value\": [${patch}]}]" --type json --subresource status &> "${catch_res_file}" || true
+    elif [[ "${action}" == "remove" ]]; then 
+        oc patch infrastructure cluster --type='json' -p '[{"op": "remove", "path": "/status/platformStatus/azure/resourceTags"}]' --subresource status &> "${catch_res_file}" || true
+    fi
+
+    if grep -q "${error}" ${catch_res_file}; then
+        echo "INFO: catch the expected error"
+    else
+        echo "ERROR: fail to catch expected error, actual error: "
+        cat "${catch_res_file}"
+        check_result=1
+    fi
+
+    rm -rf ${catch_res_file}
+}
+
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+INSTALL_CONFIG="${SHARED_DIR}/install-config.yaml"
+INFRA_ID=$(jq -r .infraID ${SHARED_DIR}/metadata.json)
+BASE_DOMAIN=$(yq-go r "${INSTALL_CONFIG}" 'baseDomain')
+BASE_DOMAIN_RG=$(yq-go r "${INSTALL_CONFIG}" 'platform.azure.baseDomainResourceGroupName')
+CLUSTER_NAME=$(yq-go r "${INSTALL_CONFIG}" 'metadata.name')
+RESOURCE_GROUP=$(yq-go r "${INSTALL_CONFIG}" 'platform.azure.resourceGroupName')
+if [[ -z "${RESOURCE_GROUP}" ]]; then
+    RESOURCE_GROUP="${INFRA_ID}-rg"
+fi
+
+check_result=0
+
+ocp_owned_key="kubernetes.io_cluster.${INFRA_ID}"
+ocp_owned_value="owned"
+user_tags_from_config=$(yq-go r ${INSTALL_CONFIG} 'platform.azure.userTags' -j | sed 's/false/"false"/g' | jq -c)
+echo "user_tags_from_config: ${user_tags_from_config}"
+#append OCP owned tag
+expected_user_tags=$(echo "${user_tags_from_config}" | jq -c -S ". += {\"${ocp_owned_key}\":\"${ocp_owned_value}\"}")
+echo "expected_user_tags: ${expected_user_tags}"
+
+#check tags on resource created in resource group
+resource_info_file=$(mktemp)
+echo "All resrouces created in cluster resource group ${RESOURCE_GROUP}:"
+az resource list -g ${RESOURCE_GROUP} --query '[].{type:type, name:name, tags:tags}' -ojson | tee ${resource_info_file}
+readarray -t res_info_array < <(jq -c '.[]' ${resource_info_file})
+echo -e "\n************ Checking resources tags created in resource group ${RESOURCE_GROUP} ************ "
+for item in "${res_info_array[@]}"; do
+    res_type=$(echo ${item} | jq -r -c '.type')
+    res_name=$(echo ${item} | jq -r -c '.name')
+    res_tags=$(echo ${item} | jq -r -c -S '.tags')
+
+    echo "(*) check tags on resource ${res_type} ${res_name}"
+    echo "tags on ${res_name}: ${res_tags}"
+
+    if [[ "${res_type}" == "Microsoft.Network/publicIPAddresses" ]] && [[ "${res_tags}" =~ k8s-azure-cluster-name ]]; then
+        echo "WARN: app public ip address is not tagged with openshift or user tags since it is not created by openshift core operator! Skip the check..."
+        continue
+    fi
+
+    if [[ "${res_type}" == "Microsoft.Compute/disks" ]] && [[ "${res_tags}" =~ kubernetes.io-created-for-pv-name ]]; then
```

Comment:
Found that pv volumes does not attach any user-defined tags and tag including clusterid, but attach below ones:
```
{"k8s-azure-created-by":"kubernetes-azure-dd","kubernetes.io-created-for-pv-name":"pvc-57b6758c-b922-4df1-977b-b5f33a3505df","kubernetes.io-created-for-pvc-name":"prometheus-data-prometheus-k8s-1","kubernetes.io-created-for-pvc-namespace":"openshift-monitoring"}
```
will check with dev if it is on purpose or something missing, and follow up in the separated PR later if requires. 
Currently, skip the checking.